synapse/lib/scrape.py

```diff
@@ -163,15 +194,31 @@ def scrape(text, ptype=None, refang=True, first=False):
             cb = opts.get('callback')
 
             for valu in regx.finditer(text):  # type: regex.Match
+                raw_span = valu.span('valu')
+                raw_valu = valu.group('valu')
+
+                info = {
+                    'ptype': ruletype,
+                    'raw_valu': raw_valu,
+                    'raw_valu_start': raw_span[0],
+                    'raw_valu_end': raw_span[1],
+                }
 
                 if cb:
-                    valu = cb(valu)
+                    # CB is expected to return a tufo of <new valu, info>
+                    valu, cbfo = cb(valu)
                     if valu is None:
                         continue
+                    new_span = cbfo.get('span')
```

Comment:
Under what circumstance will a callback provide a span that could over-ride the original span context without getting a copy of the original span boundaries?